src/ra_utility.h

```diff
@@ -21,6 +21,29 @@ to_unsigned(_In_ SignedType st) noexcept { return static_cast<std::make_unsigned
 template<typename UnsignedType, class = std::enable_if_t<std::is_unsigned_v<UnsignedType>>> _NODISCARD _CONSTANT_FN
 to_signed(_In_ UnsignedType st) noexcept { return static_cast<std::make_signed_t<UnsignedType>>(st); }
 
+/// <summary>Changes the sign of <paramref name="a" />.</summary>
+/// <typeparam name="Arithmetic">
+///   A type that satisfies the <see cref="std::is_arithmetic_v" /> constraint.
+/// </typeparam>
+/// <param name="a">Arithmetic value to change.</param>
+/// <returns>
+///   A new <typeparamref name="Arithmetic" /> with the opposite sign of <paramref name="a" />.
+/// </returns>
+/// <remarks>For floating-point numbers this just makes them either negative or positive.</remarks>
+template<typename Arithmetic, class = std::enable_if_t<std::is_arithmetic_v<Arithmetic>>> _NODISCARD _CONSTANT_FN
+flip_sign(_In_ Arithmetic a) noexcept
+{
+    if constexpr (std::is_integral_v<Arithmetic>)
+    {
+        if constexpr (std::is_signed_v<Arithmetic>)
+            return to_unsigned(a);
+        else if constexpr (std::is_unsigned_v<Arithmetic>)
+            return to_signed(a);
+    }
+    if constexpr (std::is_floating_point_v<Arithmetic>)
+        return -a;
```

Comment:
I made this convenience wrapper so we don't have to remember what the sign is. A "one-size-fits-all" function so to speak.

Reply:
This doesn't appear to be used, and has different behavior depending on if it's called with an integer or a float. 

I think it has very little benefit. I would prefer the user explicitly called `to_signed` or `to_unsigned`, or `-a` to make it clear what he's trying to do.

Reply:
That sounds good to me. Thought it would make things easier. We could just have it for Integrals because sign flipping between integer types and floating-points _are_ different. It can be gone though.